libraries/mbed/targets/hal/TARGET_WIZNET/TARGET_W7500x/i2c_api.c

```diff
@@ -155,52 +162,78 @@ int i2c_read(i2c_t *obj, int address, char *data, int length, int stop)
 
     // Read all bytes
     for (count = 0; count < (length-1); count++) {
-        if( (value = i2c_byte_read(obj, 0)) == -1) return value;
+        if( (value = i2c_byte_read(obj, 0)) == -1) {
+              return value;
+        }
         data[count] = (char)value;
     }
 
     if(stop){
-        if( (value = i2c_byte_read(obj, 1)) == -1) return value;
+        if( (value = i2c_byte_read(obj, 1)) == -1) {
+            return value;
+        }
         data[count] = (char)value;
-    }
+      
+        i2c_stop(obj);
+        obj->is_setAddress =1;
+        count++;
+      }
     else{
-        if( (value = i2c_byte_read(obj, 0)) == -1) return value;
+        if( (value = i2c_byte_read(obj, 0)) == -1) {
+     
+            return value;
+        }
         data[count] = (char)value;
+        count++;
     }
-
+    
     return count;
+    
 }
 
 int i2c_write(i2c_t *obj, int address, const char *data, int length, int stop)
 {
     I2cHandle = (I2C_TypeDef *)(obj->i2c);
-    int count;
-
+    int count =0;
+    
     if(!obj->is_setAddress)
     {
        if( I2C_Start(I2cHandle, address, I2C_WRITE_SA7) == ERROR )
         {
-            return -1;
+                    return -1;
         }
        obj->is_setAddress = 1;
        obj->ADDRESS = address;
-    }
+      }
     else
     {
         I2C_Restart_Structure(I2cHandle, address, I2C_WRITE_SA7);
         obj->ADDRESS = address;
+      
     }
 
     for (count = 0; count < length; count++) {
         i2c_byte_write(obj, data[count]);
+        wait_us(1);
     }
 
+    
+    if(length == 0x00)
+    {
```

Comment:
I2C mbed hal test code has send data 0 length.
But our I2C cannot send data 0 length so I change pin from i2c to gpio and send dummy data via i2c and change pin from gpio to i2c.